caravel/assets/visualizations/mapbox.jsx

```diff
@@ -0,0 +1,336 @@
+            ctx.fill();
+
+            if (pointLabel !== undefined) {
+              var fontHeight = d3.round(pointRadius, 1);
+              this._drawText(ctx, pixelRounded, fontHeight, pointLabel, pointRadius, rgb, false);
+            }
+          }
+        }
+      }, this);
+    }
+
+    ctx.restore();
+  }
+}
+
+class MapboxViz extends React.Component {
+  constructor(props) {
+    super(props);
+
+    var longitude = this.props.viewportLongitude || -122.405293,
+        latitude = this.props.viewportLatitude || 37.772123;
+
+    this.state = {
+      viewport: {
+        longitude: longitude,
+        latitude: latitude,
+        zoom: this.props.viewportZoom || 11,
+        startDragLngLat: [longitude, latitude]
+      }
+    };
+
+    this.onChangeViewport = this.onChangeViewport.bind(this);
+  }
+
+  onChangeViewport(viewport) {
+    this.setState({
+      viewport: viewport
+    });
+  }
+
+  render() {
+    var mercator = ViewportMercator({
+        zoom: this.state.viewport.zoom,
+        width: this.props.sliceWidth,
+        height: this.props.sliceHeight,
+        longitude: this.state.viewport.longitude,
+        latitude: this.state.viewport.latitude
+      }),
+      topLeft = mercator.unproject([0, 0]),
+      bottomRight = mercator.unproject([this.props.sliceWidth, this.props.sliceHeight]),
+      bbox = [topLeft[0], bottomRight[1], bottomRight[0], topLeft[1]],
+      clusters = this.props.clusterer.getClusters(bbox, Math.round(this.state.viewport.zoom));
+
+    d3.select("#viewport_longitude").attr("value", this.state.viewport.longitude);
+    d3.select("#viewport_latitude").attr("value", this.state.viewport.latitude);
+    d3.select("#viewport_zoom").attr("value", this.state.viewport.zoom);
+
+    return (
+      <MapGL
+        {...this.state.viewport}
+        mapStyle={this.props.mapStyle}
+        width={this.props.sliceWidth}
+        height={this.props.sliceHeight}
+        mapboxApiAccessToken={this.props.mapboxApiKey}
+        onChangeViewport={this.onChangeViewport}>
+        <ScatterPlotGlowOverlay
+          {...this.state.viewport}
+          isDragging={this.state.viewport.isDragging === undefined ? false : this.state.viewport.isDragging}
+          width={this.props.sliceWidth}
+          height={this.props.sliceHeight}
+          locations={Immutable.fromJS(clusters)}
+          dotRadius={this.props.pointRadius}
+          pointRadiusUnit={this.props.pointRadiusUnit}
+          rgb={this.props.rgb}
+          globalOpacity={this.props.globalOpacity}
+          compositeOperation={"screen"}
+          renderWhileDragging={this.props.renderWhileDragging}
+          aggregatorName={this.props.aggregatorName}
+          lngLatAccessor={function (location) {
+            const coordinates = location.get("geometry").get("coordinates");
+            return [coordinates.get(0), coordinates.get(1)];
+          }}/>
+      </MapGL>
+    );
+  }
+}
+
+function mapbox(slice) {
+  const defaultPointRadius = 60;
```

Comment:
looks like we are using a mix of `const/var`. would be great to use `const/let` only.